### ISSUE TITLE
GHC 9.2.1 compatibility

### ISSUE THREAD

**Issue by locallycompact** [stevana/quickcheck-state-machine#6]
Hi, I assume this is to do with 9.2.1, but I can't see what has changed exactly.

```
[ 2 of 21] Compiling Test.StateMachine.Lockstep.Auxiliary ( src/Test/StateMachine/Lockstep/Auxiliary.hs, dist/build/Test/State>

src/Test/StateMachine/Lockstep/Auxiliary.hs:64:13: error:
    • Could not deduce: (Top a => Elem xs a -> g a -> m (h a))
                        ~ (Elem xs a -> g a -> m (h a))
      from the context: (NTraversable f, Applicative m, SListI xs)
        bound by the type signature for:
                   ntraverse :: forall {k} (f :: (k -> *) -> [k] -> *) (m :: * -> *)
                                       (xs :: [k]) (g :: k -> *) (h :: k -> *).
                                (NTraversable f, Applicative m, SListI xs) =>
                                (forall (a :: k). Elem xs a -> g a -> m (h a))
                                -> f g xs -> m (f h xs)
        at src/Test/StateMachine/Lockstep/Auxiliary.hs:(61,1)-(63,33)
      Expected: (forall (a :: k). Elem xs a -> g a -> m (h a))
                -> f g xs -> m (f h xs)
        Actual: (forall (a :: k). Top a => Elem xs a -> g a -> m (h a))
                -> f g xs -> m (f h xs)
    • In the expression: nctraverse (Proxy @Top)
      In an equation for ‘ntraverse’: ntraverse = nctraverse (Proxy @Top)
    • Relevant bindings include
        ntraverse :: (forall (a :: k). Elem xs a -> g a -> m (h a))
                     -> f g xs -> m (f h xs)
          (bound at src/Test/StateMachine/Lockstep/Auxiliary.hs:64:1)
   |
64 | ntraverse = nctraverse (Proxy @Top)
   |             ^^^^^^^^^^^^^^^^^^^^^^^
   ```

**Comment by stevana** [comment]
Ping @edsko

**Comment by edsko** [comment]
It's due to the change in type inference introduced in 9.0. Fixed in #7.
